src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -348,6 +348,23 @@ public bool IsEquivalentTo(SyntaxNode other)
             return this.Green.IsEquivalentTo(other.Green);
         }
 
+        /// <summary>
+        /// Returns true if these two nodes are considered "incrementally identical".  An incrementally identical node
+        /// occurs when a <see cref="SyntaxTree"/> is incrementally parsed using <see cref="SyntaxTree.WithChangedText"/>
+        /// and the incremental parser is able to take the node from the original tree and use it in its entirety in the
+        /// new tree.  In this case, the <see cref="SyntaxNode.ToFullString()"/> of each node will be the same, though 
+        /// will have different parents, and may occur at different positions in their respective trees.  If two nodes are
+        /// incrementally identical, all children of each node will be incrementally identical as well.
+        /// </summary>
+        /// <remarks>
+        /// Incrementally identical nodes can also appear within the same syntax tree, or syntax trees that did not arise
+        /// from <see cref="SyntaxTree.WithChangedText"/>.  This can happen as the parser is allowed to construct parse
+        /// trees from shared nodes for efficiency.  In all these cases though, it will still remain true that the incrementally
+        /// identical nodes will have different parents and may occur at different positions in their respective trees.
```

Comment:
```suggestion
        /// identical nodes could have different parents and may occur at different positions in their respective trees.
```

Otherwise, it cannot be true that the node is incrementally identical all the way down.

Reply:
well, tehy're incrementally identical *Across* the trees.  i.e. if you were take the corresponding children in one tree and compare to those same children in the other, they would be incrementally identical, though each respective pair would have different parents.